_rules/role-required-states-and-properties-4e8ab6.md

```diff
@@ -60,47 +64,132 @@ This rule relies on browsers and assistive technologies to support leaving out [
+
+```html
+<p>My first HTML</p>
+<div role="separator"></div>
+<p>My last HTML</p>
+```
+
+#### Passed Example 6
 
-This `combobox` has required properties `aria-controls` and `aria-expanded`. `aria-controls` references an element that does not exist, but may be added to the page when expanded.
+This `combobox` has the required properties `aria-controls` and `aria-expanded`.
 
 ```html
-<div role="combobox" aria-controls="someElementId" aria-expanded="false"></div>
+<label for="tag_combo">Tag</label>
+<input type="text" id="tag_combo" role="combobox" aria-expanded="true" aria-controls="popup_listbox" />
+<ul role="listbox" id="popup_listbox">
+	<li role="option">Zebra</li>
+	<li role="option" id="selected_option">Zoom</li>
+</ul>
 ```
 
 ### Failed
 
 #### Failed Example 1
 
-This `combobox` is missing the required `aria-controls` property.
-
-**Note**: In [WAI-ARIA 1.2][], `combobox` will also require `aria-expanded`.
+This `heading` does not have the required `aria-level` property. Prior to [WAI-ARIA 1.2][] the `heading` role had an implicit default `aria-level` value of `2`. As of WAI-ARIA 1.2 this property must be explicitly set.
 
 ```html
-<div role="combobox" aria-expanded="true"></div>
+<div role="heading">
+	My First Heading
+</div>
 ```
 
 #### Failed Example 2
 
-This `combobox` has an empty value for the required `aria-controls` property.
+This `switch` does not have the required `aria-checked` property. Prior to [WAI-ARIA 1.2][] the `switch` role had an implicit default `aria-checked` value of `false`. As of WAI-ARIA 1.2 this property must be explicitly set.
+
+```html
+<div role="switch">
+	Toggle me
+</div>
+```
+
+#### Failed Example 3
+
+This `checkbox` does not have the required property `aria-checked`. Prior to [WAI-ARIA 1.2][] the `checkbox` had an implicit default `aria-checked` value of `false`. As of WAI-ARIA 1.2 this property must be explicitly set.
+
+```html
+<div role="checkbox" aria-labelledby="label"></div>
+<div id="label">Check me</div>
+```
+
+#### Failed Example 4
+
+This `separator` does not have the required `aria-valuenow` property. This is required because the `separator` is [focusable][], which makes it a `widget`.
+
+```html
+<p>My first HTML</p>
+<div role="separator" tabindex="0"></div>
+<p>My last HTML</p>
+```
+
+#### Failed Example 5
+
+This `combobox` does not have the required `aria-expanded` property. Prior to [WAI-ARIA 1.2][] the `combobox` had an implicit default `aria-expanded` value of `false`. As of WAI-ARIA 1.2 this property must be explicitly set.
+
+```html
+<label for="tag_combo">Tag</label>
+<input type="text" id="tag_combo" role="combobox" aria-controls="popup_listbox" />
+<ul role="listbox" id="popup_listbox">
+	<li role="option">Zebra</li>
+	<li role="option" id="selected_option">Zoom</li>
+</ul>
+```
+
+#### Failed Example 6
+
+This `combobox` uses `aria-owns` where it should use `aria-controls` property. This property is required as of [WAI-ARIA 1.1][].
```

Comment:
I don't really understand this description.
Does it mean that `aria-owns` used to be required on `combobox` but isn't anymore? It seems that [combobox in ARIA 1.1](https://www.w3.org/TR/wai-aria-1.1/#combobox) was already requiring `aria-controls` and `aria-expanded` (unchanged in ARIA 1.2) and wasn't required `aria-owns`.
So, I don't understand this test case (why is it needed and what does it test?)